src/o365/spo/commands/feature/feature-list.ts

```diff
@@ -0,0 +1,175 @@
+import { ContextInfo } from '../../spo';
+import Utils from '../../../../Utils';
+import { Feature } from './feature';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url: string;
+  scope?: string;
+}
+
+class SpoFeatureListCommand extends SpoCommand {
+  public get name(): string {
+    return `${commands.FEATURE_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Lists features for site or site collection';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.scope = args.options.scope || 'Web';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const scope: string = (args.options.scope) ? args.options.scope : 'Web';
+    const resource: string = Auth.getResourceFromUrl(args.options.url);
+    let siteAccessToken: string = '';
+
+    auth.getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Retrieving request digest...`);
+        }
+
+        return this.getRequestDigestForSite(args.options.url, accessToken, cmd, this.debug);
+      })
+      .then((contextResponse: ContextInfo): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(JSON.stringify(contextResponse));
+          cmd.log('');
+
+          cmd.log(`Attempt to get feature list with scope: ${scope}`);
+          cmd.log('');
+        }
+
+        const requestOptions: any = {
+          url: `${args.options.url}/_api/${scope}/Features?$select=DisplayName,DefinitionId`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            accept: 'application/json;odata=nometadata'
+          }),
+          json: true
+        };
+
+        return request.get(requestOptions);
+      })
+      .then((features: { value: Feature[] }): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(features);
+          cmd.log('');
+        }
+
+        if (features.value && features.value.length > 0) {
+          if (args.options.output === 'json') {
+            cmd.log(features.value);
+          }
+          else {
+            cmd.log(features.value.map(f => {
+              return {
+                DefinitionId: f.DefinitionId,
+                DisplayName: f.DisplayName
+              };
+            }));
+          }
+        }
+        else {
+          if (this.verbose) {
+            cmd.log('No features found');
+          }
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedPromise(err, cmd, cb));
```

Comment:
Because you set `json:true` in the request, you should use `handleRejectedODataJsonPromise` to handle the rejected promise